pymongo/errors.py

```diff
@@ -243,6 +243,10 @@ def __init__(self, results):
         # For pickle support
         self.args = (results,)
 
+    if sys.version_info[0] != 2:
+        def __str__(self):
+            return str(self._message)
+
```

Comment:
The fact that we have to add `__str__` to fix this bug makes me a little uncomfortable. It actually makes me think that using `self.args = (results,)` to "fix" the original pickle issue (in https://github.com/mongodb/mongo-python-driver/pull/514) was the wrong choice.

Looking again, I think the best option may be to use `__reduce__` like this:

```suggestion

    def __reduce__(self):
        return self.__class__, (self.details,)

```

Could you make the above change?

Reply:
Way better. I took your recommendation.